app/httphandler_dump.go

```diff
@@ -0,0 +1,146 @@
+package app
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"net/http"
+	"runtime"
+	"runtime/pprof"
+)
+
+func dumpGoroutines(w http.ResponseWriter, r *http.Request) {
+	var b [4 * 1024 * 1024]byte
+
+	n := runtime.Stack(b[:], true)
+	if n > 0 {
+		w.Write(b[0:n])
+	} else {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("This is not the DUMP you are looking for."))
```

Comment:
Nit: maybe these "This is not the DUMP (...)"  messages could be changed to something with a hint that an error occurred. The current messages gives more a vibe of "access error" imo